### PR TITLE
fixes some oddities in addiction code

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
fix: Fixes being on a drug not stopping the effects of withdrawal when you're addicted
/:cl:

also reverts some of armhulens changes, they didn't actually fix most of the issues with addiction and made some of the code worse while changing balance.

when writing the code for addiction it seems I slapped my head against the desk sometime and forgot to account for the fact that the addictions need to be reset when you start taking the drug again. this is now fixed.

### CODE REVIEW

**Comment by Ghilker** [comment]
he didn't `FIX EVERYTHING`

**Comment by Time-Green** [comment]
please... speedmerge... i cant take it anymore

**Comment by Qustinnus** [comment]
owned noob

File: code/modules/reagents/withdrawal/generic_addictions.dm [hunk 1]
```diff
@@ -83,12 +83,12 @@
 	. = ..()
 	affected_carbon.apply_status_effect(/datum/status_effect/trance, 40 SECONDS, TRUE)
 
-/datum/addiction/hallucinogens/lose_addiction(datum/mind/victim_mind)
+/datum/addiction/hallucinogens/end_withdrawal(mob/living/carbon/affected_carbon)
 	. = ..()
-	var/atom/movable/plane_master_controller/game_plane_master_controller = victim_mind.current.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+	var/atom/movable/plane_master_controller/game_plane_master_controller =affected_carbon.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
```
Comment (Mothblocks) [inline code/modules/reagents/withdrawal/generic_addictions.dm:88]:
```suggestion
	var/atom/movable/plane_master_controller/game_plane_master_controller = affected_carbon.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
```

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**